pyiron: Require pyiron_base 0.1.16

jan-janssen opened the pull request.

bgruening commented on pyiron/pyiron.xml:
```diff
@@ -2,6 +2,7 @@
     <description>meta tools</description>
     <requirements>
         <requirement type="package" version="0.3.2">pyiron</requirement>
+        <requirement type="package" version="0.1.16">pyiron_base</requirement>
```
why is this needed, shouldn't pyiron not get the pyiron_base package?

jan-janssen commented:
It does, but I just updated pyiron_base to 0.1.16 previously it was 0.1.15 so to enforce the installation of pyiron_base 0.1.16 I added it as explizit dependency.

bgruening commented:
I have fixed this locally. Thanks.
